src/workerd/jsg/resource.h

```diff
@@ -1411,9 +1460,17 @@ class ResourceWrapper {
     // type as the global, we instantiate this separate branch specifically for that type.
     // Fortunately, for types that are never used as the global object, we never have to
     // instantiate the `isContext = true` branch.
+    //
+    // When creating a new context the global context template will be memoized for future reuse in
+    // additional contexts. The global context can depend on `configuration`. For Python Isolate
+    // Pools we want to create a new context and initialize some code in it and later when a request
+    // arrives, edit the configuration and create a new context. To achieve this we create the first
+    // context with `memoize=false` to disable memoization of the global context created for the
+    // first context. Then we create a new context with `memoize=true` to enable memoization of the
+    // main context created after a request was received.
```

Comment:
Probably a stupid question, but I wonder why we cannot overwrite the memoized context after it has been memoized?

Reply:
That would require the inverse of the same logic changing the code from that existed before this change rather than adding a different code flow for our exceptional case.